app/Jobs/ImportTurboVotePosts.php

```diff
@@ -340,5 +320,63 @@ private function scrubRecord($record)
 
         return $isNotValidEmail || $isNotValidHostname || $isNotValidLastName ? false : true;
     }
+
+    /**
+     * For a given record and referral code values, first check if we have a northstar ID, then grab the user using that.
+     * Otherwise, see if we can find the user with the given email (if it exists), if not check if we can find them with the given phone number (if it exists).
+     * If all else fails, create the user .
+     *
+     * @TODO - If we have a northstar id in the referral code, then we probably don't need to make a call to northstar for the full user object.
+     *
+     * @return array
+     */
+    private function getOrCreateUser($record, $values)
+    {
+        $user = null;
+
+        if (isset($values['northstar_id']) && !empty($values['northstar_id'])) {
+            $this->stats['countHasNorthstarID']++;
+
+            $user = gateway('northstar')->asClient()->getUser('id', $values['northstar_id']);
+
+            return $user;
+        }
+
+        $this->stats['countMissingNSId']++;
+
+        if (isset($record['email']) &&  !empty($record['email'])) {
+            $user = gateway('northstar')->asClient()->getUser('email', $record['email']);
+        } elseif (isset($record['phone']) && !empty($record['phone']) ) {
+            $user = gateway('northstar')->asClient()->getUser('mobile', $record['phone']);
+        } else {
+            $user = gateway('northstar')->asClient()->createUser([
+                'email' => $record['email'],
+                'mobile' => $record['phone'],
+                'first_name' => $record['first-name'],
+                'last_name' => $record['last-name'],
+                'addr_street1' => $record['registered-address-street'],
+                'addr_street2' => $record['registered-address-street-2'],
+                'addr_city' => $record['registered-address-city'],
+                'addr_state' => $record['registered-address-state'],
+                'addr_zip' => $record['registered-address-zip'],
```

Comment:
Do you need to check if these fields are set like you do in all the conditionals above?

Reply:
these aren't required. so i didn't do the extra check.